### ISSUE TITLE
swc-angular-preset: Expression expected/syntax error because of decorator in angular

### ISSUE THREAD

**Issue by LanderBeeuwsaert** [jscutlery/devkit#302]
We just tried moving from plain jest-preset-angular to swc-angular-preset.
However it seems that somehow the library doesn't agree with the tests we have, which reference classes with (angular) decorators:

We get a lot of tests that fail with the following message:
![image](https://github.com/jscutlery/devkit/assets/1098243/3b43387b-d580-4c67-a46d-375de654b757)

**Comment by yjaaidi** [comment]
Thanks @LanderBeeuwsaert for raising the issue.
This is surprising. This usually happens when decorators are disabled in SWC (i.e. `jsc.parser.decorators: false`).
Are using the SWC preset like this in your jest config:
```ts
    '^.+\\.(ts|mjs|js)$': [
      '@swc/jest',
      swcAngularPreset
    ],
 ```
 or did you customize the SWC configuration?
 
 Could you provide a repro?

 Thank you in advance.

**Comment by LanderBeeuwsaert** [comment]
Hi @yjaaidi , thanks for the quick response.

Our config has a bit of changes that have been added over the years, but even when commenting out those I still get the error.
I'm attaching a zip with our setup. Providing a repro will take me considerably more time so I'll look for finding time to do that if you would be able to confirm that from just the files it's not easy to directly see what we are doing wrong.

weird thing is that with the same files/setup and jest-preset-angular it does work (but just waaaaaay slower :-D)

also, is there an example repo with a setup that works? In that case I would be able to clone that and work step by step myself finding the configuration item that makes it break.

**Comment by LanderBeeuwsaert** [comment]
[config.zip](https://github.com/jscutlery/devkit/files/14864382/config.zip)

**Comment by yjaaidi** [comment]
Indeed, this is all very surprising. There doesn't seem to be anything wrong with your configuration.
Even though I can't see how this could be related, can you try the following troubleshooting steps just to make sure it's not a side effect of another problem:
- clear jest cache: `jest --clearCache`
- update the plugin `npm i -D @jscutlery/swc-plugin-angular@0.10.1`

You can find working examples in our repo 😉: https://github.com/jscutlery/devkit/tree/main/tests/swc-plugin-angular-wide

**Comment by LanderBeeuwsaert** [comment]
@yjaaidi voilà, this was the simplest repro I could do, and I still have the same issue
Which I didn't expect to happen actually.
I'm sure I'm missing something obvious

https://github.com/LanderBeeuwsaert/swc-jest-test

![image](https://github.com/jscutlery/devkit/assets/1098243/370a45b5-2368-430f-b43c-876e2e066f55)

**Comment by yjaaidi** [comment]
Thank you so much @LanderBeeuwsaert for the repro.

There was a problem with the 0.1.2 bundle (the default export was invalid) which meant that the config was `undefined` and SWC was falling back to its default options that didn't transform decorators. 
This was fixed (at least since 0.2.0).

Also, switching to the newly recommended way of configuring Jest is safer: https://github.com/jscutlery/devkit/tree/main/packages/swc-angular-preset#2-configure

```ts
import { swcAngularJestTransformer } from '@jscutlery/swc-angular-preset';

...
   '^.+\\.(ts|mjs|js)$': swcAngularJestTransformer(),
...
```

Please let us know how this works out for you.

Thank you in advance!

**Comment by LanderBeeuwsaert** [comment]
@yjaaidi amazing! works like a charm: man it now runs in 10% of the time as before. That's crazy. Thanks! Closing
